mesonpy/__init__.py

```diff
@@ -267,166 +301,34 @@ def _debian_python(self) -> bool:
-        """Map files into ABI tags."""
-        files_by_tag: Dict[mesonpy._tags.Tag, List[str]] = collections.defaultdict(list)
-
-        for _, file in self._wheel_files['platlib']:
-            # if in platlib, calculate the ABI tag
-            tag = self._calculate_file_abi_tag_heuristic(file)
-            if tag:
-                files_by_tag[tag].append(file)
-
-        return files_by_tag
-
-    def _select_abi_tag(self) -> Optional[mesonpy._tags.Tag]:  # noqa: C901
-        """Given a list of ABI tags, selects the most specific one.
-
-        Raises an error if there are incompatible tags.
+        All files that look like extension modules are verified to
+        have a file name compatibel with what is expected by the
+        Python interpreter. An exception is raised otherwise.
+
+        Other files are ignored.
         """
-        # Possibilities:
-        #   - interpreter specific (cpython/pypy/etc, version)
-        #   - stable abi (abiX)
-        tags = self._files_by_tag()
-        selected_tag = None
-        for tag, files in tags.items():
-            # no selected tag yet, let's assign this one
-            if not selected_tag:
-                selected_tag = tag
-            # interpreter tag
-            elif isinstance(tag, mesonpy._tags.InterpreterTag):
-                if tag != selected_tag:
-                    if isinstance(selected_tag, mesonpy._tags.InterpreterTag):
-                        raise ValueError(
-                            'Found files with incompatible ABI tags:\n'
-                            + self._file_list_repr(tags[selected_tag])
-                            + '\tand\n'
-                            + self._file_list_repr(files)
-                        )
-                    selected_tag = tag
-            # stable ABI
-            elif isinstance(tag, mesonpy._tags.StableABITag):
-                if isinstance(selected_tag, mesonpy._tags.StableABITag) and tag != selected_tag:
-                    raise ValueError(
-                        'Found files with incompatible ABI tags:\n'
-                        + self._file_list_repr(tags[selected_tag])
-                        + '\tand\n'
-                        + self._file_list_repr(files)
-                    )
-        return selected_tag
+        abis = []
+
+        for path, src in self._wheel_files['platlib']:
+            if os.name == 'nt' and path.suffix == '.pyd' or path.suffix == '.so':
+                name, suffix = path.name.split('.', 1)
+                if suffix not in _EXTENSION_SUFFIXES:
+                    raise ValueError('Extension module "{}" not compatible with Python interpreter.'.format(path))
+                match = _EXTENSION_SUFFIX_REGEX.match(suffix)
+                if match is None:
+                    raise ValueError('Suffix "{}" for extension module "{}" not understood.'.format(suffix, path))
+                abis.append(match.group('abi'))
+
+        return all(abi == 'abi3' for abi in abis)
```

Comment:
We should change this function to return the ABI version.

Reply:
There is only one versioned ABI tag. See the other comment.